.github/workflows/zeebe-ci.yml

```diff
@@ -37,9 +37,11 @@ on:
   workflow_dispatch: { }
   workflow_call: { }
 
+# Limit workflow to 1 concurrent run per ref (branch): new commit -> old runs are canceled to save costs
+# Exception for main branch: complete builds for every commit needed for confidenence
 concurrency:
   cancel-in-progress: true
-  group: "${{ github.workflow }}-${{ github.ref }}"
+  group: ${{ github.ref == 'refs/heads/main' && github.sha || format('{0}-{1}', github.workflow, github.ref) }}
```

Comment:
:x: This workflow deploys snapshot artifacts if it runs on main, see https://github.com/camunda/camunda/blob/ebd6677b941cd64e8e8107bb37b9bbb7979ee882/.github/workflows/zeebe-ci.yml#L601-L605. Therefor it is important that it does not run concurrently on main.

:x: Can we not achieve the same by saying `cancel-in-progress: ${{ github.ref != 'refs/heads/main'}}`? That way we cancel in-progress ones for PRs (saving resources) while queuing all runs on main, giving us safety for deploying snapshot artifacts _and_ increased test coverage.

Reply:
> Therefor it is important that it does not run concurrently on main.

Gotcha, this is a very good point! I'm not insisting on having this change in all workflows so I it is fine for me to drop it for this workflow 🙂 

> 💭 Can we not achieve the same by saying `cancel-in-progress: ${{ github.ref != 'refs/heads/main'}}`? That way we cancel in-progress ones for PRs (saving resources) while queuing all runs on main, giving us safety for deploying snapshot artifacts and increased test coverage.

Maaaaybe, I'm not 110% sure if that's possible and there are no other downsides. Going back to status quo before this PR for this 1 workflow is fine as well

Reply:
> Maaaaybe, I'm not 110% sure if that's possible and there are no other downsides. Going back to status quo before this PR for this 1 workflow is fine as well

Happy to discuss this, I currently don't see any risk with this approach. Assuming it works as intended, I think this would be the _safest_ option, especially because its future proof when one of the other workflows gets extended to also deploy artifacts or something similar.

Reply:
Actually it also seems to be possible to do this on the job level as demonstrated here by creating a different concurrency group: https://github.com/camunda/camunda/blob/edd4af9c3a512982b11ff1a5380c83c7184d3b13/.github/workflows/ci.yml#L490-L492

That should work together with not cancelling builds on `main` because the *deploy jobs specifically would wait on each other 🤔

Reply:
Zeebe CI already has that: https://github.com/camunda/camunda/blob/edd4af9c3a512982b11ff1a5380c83c7184d3b13/.github/workflows/zeebe-ci.yml#L607-L609 🤯 Sooo all good with my original change proposal?

Reply:
> But that's not a strong opinion, feel free to merge as is.

I agree and will keep it in mind for future change but right now will not do it to get the original purpose of the PR merged.

Reply:
I'll also want to document the behavior at one point properly to save time for future-selves.